### PR TITLE
fix: messaging settings and url unfurling fixes

### PR DESCRIPTION
**Pull request by igor-sirotin**
Closes https://github.com/status-im/status-desktop/issues/12397
Closes https://github.com/status-im/status-desktop/issues/10864
Closes https://github.com/status-im/status-desktop/issues/11604

### What does the PR do

1. Remove old unfurling methods, settings and UI
2. Remove "open in status" browser setting, force `false` value
4. Moved `History nodes` section to `Advanced`
5. Replace GIF setting `isTenorWarningAccepted` with `gifUnfurlingEnabled`
6. Refactor `localUnfurlLinks` property, use simple `gifLinks` instead
7. Filter out GIF links when calling `UnfurlURLs` endpoint
8. Create a single "Ask to allow unfurling GIFs" item when GIF unfurling is disabled
9. FIx the issue that `updateLinkPreviews` was never called in QML.

NOTE: Can (almost) be reviewed commit by commit. Just the latest commit apples a bunch f fixes.

https://github.com/status-im/status-desktop/assets/25482501/707e7545-c42d-4c73-9959-666ddbe6c708

### CODE REVIEW

**Review by caybro — COMMENTED** [review]
Code LGTM

Just wondering... we have some additional code in AppMain.qml that also manipulates `localAccountSensitiveSettings.whitelistedUnfurlingSites`. Shouldn't that be also taken care of?

**Comment by alexjba** [comment]
Looks like a legit squish test failure:
```
Exception: Timeout reached while validating: Scrolling settingsContentBase_ScrollView until linkPreviewSwitch is visible (Screenshot in "/home/jenkins/workspace/_linux_x86_64_tests-e2e_PR-12457/squish_1_782/webreport_231016_174150.725_linux-01/data/suite_communities/tst_communityMessageFlows/errorImages/error_3.png")
```

**Comment by igor-sirotin** [comment]
> Looks like a legit squish test failure:
> 
> ```
> Exception: Timeout reached while validating: Scrolling settingsContentBase_ScrollView until linkPreviewSwitch is visible (Screenshot in "/home/jenkins/workspace/_linux_x86_64_tests-e2e_PR-12457/squish_1_782/webreport_231016_174150.725_linux-01/data/suite_communities/tst_communityMessageFlows/errorImages/error_3.png")
> ```

yep, @anastasiyaig helped me with that 👌

**Comment by anastasiyaig** [comment]
> Looks like a legit squish test failure:
> 
> ```
> Exception: Timeout reached while validating: Scrolling settingsContentBase_ScrollView until linkPreviewSwitch is visible (Screenshot in "/home/jenkins/workspace/_linux_x86_64_tests-e2e_PR-12457/squish_1_782/webreport_231016_174150.725_linux-01/data/suite_communities/tst_communityMessageFlows/errorImages/error_3.png")
> ```

yes, @alexjba i will address the UI change made in this PR later, when it lands to master . For now, i just skipped the test but i will bring it back soon.

File: ui/imports/shared/views/chat/LinksMessageView.qml [hunk 1]
```diff
@@ -31,55 +31,24 @@ Flow {
...
-            readonly property bool isUserProfileLink: link.toLowerCase().startsWith(Constants.userLinkPrefix.toLowerCase())
-            readonly property string thumbnailUrl: result && result.thumbnailUrl ? result.thumbnailUrl : ""
-            readonly property string title: result && result.title ? result.title : ""
-            readonly property string hostname: result && result.site ? result.site : ""
-            readonly property bool animated: isImage && result.contentType === "image/gif" // TODO support more types of animated images?
+    Loader {
+        visible: root.gifLinks && root.gifLinks.length > 0
+                 && !RootStore.gifUnfurlingEnabled
+                 && d.localAskAboutUnfurling && !RootStore.neverAskAboutUnfurlingAgain
+        sourceComponent: enableLinkComponent
```
Comment (alexjba) [inline ui/imports/shared/views/chat/LinksMessageView.qml:45]: The `enableLinkComponent` is not working. The enable link preview component is not visible when GIFs are disabled from Settings, instead probably the Loader keeps the previous size and the message row height is too large.

Screenshot 2023-10-17 at 13 26 11

File: ui/imports/shared/views/chat/MessageView.qml [hunk 2]
```diff
@@ -66,19 +66,16 @@ Loader {
...
         const filtered = arr.filter(value => {
-                                        const v = value.toLowerCase()
-                                        return localAccountSensitiveSettings.gifUnfurlingEnabled && value.endsWith('.gif')
+                                        return value.toLowerCase().endsWith('.gif')
                                     })
-
-
-        const out = filtered.join(separator)
-        return out
+        console.log("<<< calculating gifLinks:", filtered)
```
Comment (alexjba) [inline ui/imports/shared/views/chat/MessageView.qml]: Leftovers?

**Review by alexjba — COMMENTED** [review]
Nice cleanup!

I have a few minor findings and questions, but the only one I think it needs to be fixed here is an issue where hiding GIFs from settings leaves a large empty space in the message row

Screenshot 2023-10-17 at 13 26 11

File: ui/app/AppLayouts/Profile/views/MessagingView.qml
[the remark this answers is not in the document]
Comment (igor-sirotin) [reply ui/app/AppLayouts/Profile/views/MessagingView.qml:1]: I thought to keep it for the future. We have plans to use it someday.

File: src/backend/privacy.nim [hunk 3]
```diff
@@ -17,7 +17,3 @@ proc changeDatabasePassword*(keyUID: string, oldHashedPassword: string, newHashe
   except RpcException as e:
     error "error", methodName = "changeDatabasePassword", exception=e.msg
     raise newException(RpcException, e.msg)
-
-proc getLinkPreviewWhitelist*(): RpcResponse[JsonNode] {.raises: [Exception].} =
-  let payload = %* []
-  result = callPrivateRPC("getLinkPreviewWhitelist".prefix, payload)
```
[the remark this answers is not in the document]
Comment (igor-sirotin) [reply src/backend/privacy.nim:23]: https://github.com/status-im/status-go/issues/4162

File: storybook/pages/StatusChatInputPage.qml [hunk 4]
```diff
@@ -64,7 +64,7 @@ SplitView {
         Component.onCompleted: {
             RootStore.isGifWidgetEnabled = true
```
[the remark this answers is not in the document]
Comment (igor-sirotin) [reply storybook/pages/StatusChatInputPage.qml:65]: I'll remove it in this PR 👌

File: ui/imports/shared/views/chat/MessageView.qml (hunk 2 above)
Comment (igor-sirotin) [reply ui/imports/shared/views/chat/MessageView.qml]: removed in 531a8cb1215bcc377bf37966ebdb26bc20880589 🙂

File: ui/imports/shared/views/chat/LinksMessageView.qml (hunk 1 above)
Comment (igor-sirotin) [reply ui/imports/shared/views/chat/LinksMessageView.qml:50]: oggghh, this was a tough one.

https://github.com/status-im/status-desktop/assets/25482501/36e8fba5-b75b-4cd3-b4b6-a7a733eaa908

The missing line turned out to be this one:
https://github.com/status-im/status-desktop/blob/452079d4e0e0578f31dce32c3dc0c3520dafb0e7/ui/StatusQ/src/StatusQ/Components/StatusMessage.qml#L360

I also changed the logic inside `LinksMessageView`, but that wasn't enough.

For some reason, when we take an item that is inside a `Loader`, and change it's `height` to `0`, the `Loader`'s height is automatically kept to be the initial `Loader.height` value.
`0` is a special value here. Changing it to ANY other value, e.g. even `1`  works, but setting item's `height` to `0` doesn't.

____
NOTE: For testing I hardcoded link previews into storybook's `StatusMessagePage` (you can see that in thevideo). But I don't think it makes sense to commit it, it's will take too much time to make it clean, I don't think it's worth it.

**Review by alexjba — APPROVED** [review]
Very nice! Thank you for the cleanup! 💯

**Review by caybro — APPROVED** [review]
LGTM

**Pull request merged**